components/Buttons/src/MDCFloatingButton.m

```diff
@@ -71,55 +112,171 @@ - (instancetype)initWithFrame:(CGRect)frame shape:(MDCFloatingButtonShape)shape
+    contentSize.height = MAX(self.minimumSize.height, contentSize.height);
+  }
+  if (self.maximumSize.height > 0) {
+    contentSize.height = MIN(self.maximumSize.height, contentSize.height);
+  }
+  if (self.minimumSize.width > 0) {
+    contentSize.width = MAX(self.minimumSize.width, contentSize.width);
+  }
+  if (self.maximumSize.width > 0) {
+    contentSize.width = MIN(self.maximumSize.width, contentSize.width);
   }
+  return contentSize;
 }
 
 - (void)layoutSubviews {
   // We have to set cornerRadius before laying out subviews so that the boundingPath is correct.
   self.layer.cornerRadius = CGRectGetHeight(self.bounds) / 2;
   [super layoutSubviews];
+
+   if (self.mode == MDCFloatingButtonModeNormal) {
+    return;
+  }
+
+  // Position the imageView and titleView
+  //
+  // +------------------------------------+
+  // |    |  |  |  CEI TOP            |   |
+  // |CEI +--+  |       +-----+       |CEI|
+  // | LT ||SP|-- A --|Title|-- A --|RGT|
+  // |    +--+  |       +-----+       |   |
+  // |    |  |  |  CEI BOT            |   |
+  // +------------------------------------+
+  //
+  // (A) The same spacing on either side of the label.
+  // (SP) The spacing between the image and title
+  // (CEI) Content Edge Insets
+  //
+  // The diagram above assumes an LTR user interface orientation
+  // and a .leadingIcon imageLocation for this button.
+
+  BOOL isLeadingIcon = self.imageLocation == MDCFloatingButtonImageLocationLeading;
+  UIEdgeInsets adjustedLayoutInsets =
+      isLeadingIcon ? internalLayoutSpacingInsets
+                    : UIEdgeInsetsFlippedHorizonally(internalLayoutSpacingInsets);
+
+  const CGRect insetBounds = UIEdgeInsetsInsetRect(UIEdgeInsetsInsetRect(self.bounds,
+                                                                         adjustedLayoutInsets),
+                                                   self.contentEdgeInsets);
+
+  const CGFloat imageViewWidth = CGRectGetWidth(self.imageView.bounds);
+  const CGFloat boundsCenterY = CGRectGetMidY(insetBounds);
+  CGFloat titleWidthAvailable = CGRectGetWidth(insetBounds);
+  titleWidthAvailable -= imageViewWidth;
+  titleWidthAvailable -= self.imageTitleSpacing;
+
+  const CGFloat availableHeight = CGRectGetHeight(insetBounds);
+  CGSize titleIntrinsicSize
+      = [self.titleLabel sizeThatFits:CGSizeMake(titleWidthAvailable, availableHeight)];
+
+  const CGSize titleSize = CGSizeMake(MAX(0, MIN(titleIntrinsicSize.width, titleWidthAvailable)),
+                                      MAX(0, MIN(titleIntrinsicSize.height, availableHeight)));
+
+  CGPoint titleCenter;
+  CGPoint imageCenter;
+  BOOL isRTL = self.mdf_effectiveUserInterfaceLayoutDirection
+      == UIUserInterfaceLayoutDirectionRightToLeft;
+
+  // isRTL && isLeadingIcon => icon to right of text
+  // !isRTL && !isLeadingIcon => icon to right of text
+  // isRTL && !isLeadingIcon => icon to left of text
+  // !isRTL && isLeadingIcon => icon to left of text
+  if (isRTL == isLeadingIcon) {
```

Comment:
Consider being a bit more verbose in favor of obvious, and also consider prioritizing the commonly-used path:

// If we are in Left-To-Right and the icon is leading -OR- if we are in Right-To-Left and the icon is trailing
// layout the image to the left of the label.  Otherwise layout the image to the right of the label.
if ( (ifLTR && isLeadingIcon) ||
      (!ifLTR && !isLeadingIcon) {
   // Layout image to the left of the label
   ....
} else {
   // Layout image to the right of the label
   ....
}

Reply:
Done